censys-subdomain-finder.py

```diff
@@ -10,95 +14,131 @@
+    print(
+        "[*] Found %d unique subdomain%s of %s in ~%s seconds\n"
+        % (
+            len(subdomains),
+            "s" if len(subdomains) > 1 else "",
+            domain,
+            str(time_elapsed),
+        )
+    )
     for subdomain in subdomains:
-        print('  - ' + subdomain)
+        print("  - " + subdomain)
+
+    print("")
 
-    print('')
 
 # Saves the list of found subdomains to an output file
 def save_subdomains_to_file(subdomains, output_file):
     if output_file is None or len(subdomains) == 0:
         return
 
     try:
-        with open(output_file, 'w') as f:
+        with open(output_file, "w") as f:
             for subdomain in subdomains:
-                f.write(subdomain + '\n')
+                f.write(subdomain + "\n")
 
-        print('[*] Wrote %d subdomains to %s' % (len(subdomains), os.path.abspath(output_file)))
+        print(
+            "[*] Wrote %d subdomains to %s"
+            % (len(subdomains), os.path.abspath(output_file))
+        )
     except IOError as e:
-        sys.stderr.write('[-] Unable to write to output file %s : %s\n' % (output_file, e))
+        sys.stderr.write(
+            "[-] Unable to write to output file %s : %s\n" % (output_file, e)
+        )
+
 
 def main(domain, output_file, censys_api_id, censys_api_secret, limit_results):
-    print('[*] Searching Censys for subdomains of %s' % domain)
+    print("[*] Searching Censys for subdomains of %s" % domain)
     start_time = time.time()
-    subdomains = find_subdomains(domain, censys_api_id, censys_api_secret, limit_results)
+    subdomains = find_subdomains(
+        domain, censys_api_id, censys_api_secret, limit_results
+    )
     subdomains = filter_subdomains(domain, subdomains)
     end_time = time.time()
-    time_ellapsed = round(end_time - start_time, 1)
-    print_subdomains(domain, subdomains, time_ellapsed)
+    time_elapsed = round(end_time - start_time, 1)
+    print_subdomains(domain, subdomains, time_elapsed)
     save_subdomains_to_file(subdomains, output_file)
 
+
 if __name__ == "__main__":
     args = cli.parser.parse_args()
 
     censys_api_id = None
     censys_api_secret = None
 
-    if 'CENSYS_API_ID' in os.environ and 'CENSYS_API_SECRET' in os.environ:
-        censys_api_id = os.environ['CENSYS_API_ID']
-        censys_api_secret = os.environ['CENSYS_API_SECRET']
+    if "CENSYS_API_ID" in os.environ and "CENSYS_API_SECRET" in os.environ:
+        censys_api_id = os.environ["CENSYS_API_ID"]
+        censys_api_secret = os.environ["CENSYS_API_SECRET"]
 
     if args.censys_api_id and args.censys_api_secret:
         censys_api_id = args.censys_api_id
         censys_api_secret = args.censys_api_secret
 
     limit_results = not args.commercial
     if limit_results:
-        print('[*] Applying non-commerical limits (' + str(NON_COMMERCIAL_API_LIMIT) + ' results at most)')
+        print("[*] Applying non-commercial limits (1000 results at most)")
```

Comment:
ideally should be in a variable

Reply:
Totally, we just have to break out pages/results per page, how would you like to do that?

Reply:
no strong opinion, I would go with the max page size unless there's a reason to use something else